pkg/integrations/v4/dm/emitter.go

```diff
@@ -140,13 +135,56 @@ func (e *emitter) process(
 		e.metricsSender.SendMetrics(metrics)
 	}
 
+	registeredEntities, unregisteredEntitiesWithWait := e.RegisterEntities(entities)
+
+	for entityName, entityID := range registeredEntities {
+		processEntityDataset(datasetsByEntityName[entityName], entityID)
+	}
+
+	if len(unregisteredEntitiesWithWait.entities) == 0{
```

Comment:
here we return error in case that all the entities were successfully submitted?

Reply:
We have no error handling in that function
```
	if len(emitErrs) == 0 {
		return nil
	}
```